app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -215,6 +232,22 @@ def docker_auth
     @ems.authentications.find_by(:authtype => "smartstate_docker")
   end
 
+  # To run SSA, VPC needs to turn on enableDnsSupport and enableDnsHostnames
+  def get_dns_enabled_vpcs
+    ec2.client.describe_vpcs.vpcs.select do |vpc|
+      id = vpc.vpc_id
+      enabled_dns_support?(id) && enabled_dns_hostnames?(id)
+    end
+  end
+
+  def enabled_dns_hostnames?(vpc_id)
```

Comment:
Wouldn't it use fewer API calls if you pass a VPC object in to this method?  Then you wouldn't need to do the initial lookup here

Reply:
Yes, changed accordingly.